zcash_client_sqlite/src/lib.rs

```diff
@@ -224,6 +225,39 @@ impl<P: consensus::Parameters + Clone> WalletDb<Connection, P> {
     }
 }
 
+impl<C: Borrow<rusqlite::Connection>, P: consensus::Parameters> WalletDb<C, P> {
+    pub fn read_transactionally<F, A, E: From<rusqlite::Error>>(&self, f: F) -> Result<A, E>
+    where
+        F: FnOnce(&WalletDb<SqlTransaction<'_>, P>) -> Result<A, E>,
+    {
+        let mut rng = thread_rng();
+        for backoff_ms in [0u64, 1, 2, 4, 8, 16, 32].into_iter().chain(repeat(64)) {
+            match self.conn.borrow().unchecked_transaction() {
+                Ok(tx) => {
+                    let wdb = WalletDb {
+                        conn: SqlTransaction(&tx),
+                        params: self.params.clone(),
+                    };
+                    let result = f(&wdb)?;
```

Comment:
Huh, why did the compiler allow `f` to be called in a loop when its type is only known to be `FnOnce`? (I intended `Fn`, but the question stands as to how this currently compiles.)